src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_bool.cs

```diff
@@ -12,31 +12,88 @@ public static partial class InvariantUtf16
         {
             public unsafe static bool TryParseBoolean(char* text, int length, out bool value)
             {
-                int consumed;
-                var span = new ReadOnlySpan<byte>(text, length * 2);
-                return PrimitiveParser.TryParseBoolean(span, out value, out consumed, EncodingData.InvariantUtf16);
+                ReadOnlySpan<char> span = new ReadOnlySpan<char>(text, length);
```

Comment:
Doesn't this defeat he purpose of the overload? It'll be slower with the slow span I'd assume